### ISSUE TITLE
singleAttributePerLine support number

### ISSUE THREAD

**Issue by eleven-net-cn** [prettier/prettier#14668]
![image](https://user-images.githubusercontent.com/20358391/230596797-833d733f-75d0-4c1b-b1b0-948c41f2929c.png)

The options `singleAttributePerLine` only supports to set `true/false`, It would be nice if we could set `number` to specify the number of attributes.

**Comment by eleven-net-cn** [comment]
`**-**-per-line` is turned off by default in [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier).

- https://github.com/prettier/eslint-config-prettier/blob/main/index.js#L124
- https://github.com/prettier/eslint-config-prettier/blob/main/index.js#L126
- https://github.com/prettier/eslint-config-prettier/blob/main/index.js#L154

Make it impossible to customize the max attributes of each row, None of the following configurations takes effect:

react

```js
{
  "rules": {
    "react/jsx-max-props-per-line": ["error", { "maximum": 3 }]
  }
}
```

vue

```js
{
  "rules": {
    "vue/max-attributes-per-line": [
      "error", {
        "singleline": { "max": 3 }}]
  }
}
```

So that, expect to be able to set the maximum number of attributes per-line.

Currently, if add `vue/max-attributes-per-line` with `"singleline": { "max": 3 }`, It will be invalid due to prettier's rule `printWidth` (e.g. `printWidth: 120`).

**Comment by fisker** [comment]
We don't consider compatibility with ESLint rules.

**Comment by eleven-net-cn** [comment]
> We don't consider compatibility with ESLint rules.

But the offical provides `prettier/eslint-plugin-prettier` `prettier/eslint-config-prettier`, When a developer tries to use prettier as a plugin for eslint, we need to consider these situations.

**Comment by fisker** [comment]
No, they disable rules conflicts with Prettier, but Prettier don't care which rules conflicts with us.

**Comment by eleven-net-cn** [comment]
> No, they disable rules conflicts with Prettier, but Prettier don't care which rules conflicts with us.

It seems that we should not use eslint and prettier at the same time.

**Comment by eleven-net-cn** [comment]
Hello, is there any conclusion?
